File: server_data/contests/Contest3/replay/generator.py
from random import randint, shuffle
from string import ascii_lowercase
import sys

is_pal = randint(0, 1)

n = randint(1, 10 ** 2)
if not is_pal:
    cnts = [0] * 26
    for _ in range(n):
        cnts[randint(0, 25)] += 1

else:
    cnts = [0] * 26
    for _ in range(n // 2):
        cnts[randint(0, 25)] += 2

    if n % 2 == 1:
        cnts[randint(0, 25)] += 1

print(*cnts)

ans = ''
mid = ''
for i in range(26):
    if cnts[i] % 2 == 1:
        if mid:
            print(-1, file=sys.stderr)
            break

        mid = ascii_lowercase[i]

    ans += ascii_lowercase[i] * (cnts[i] // 2)

else:
    ans = list(ans)
    shuffle(ans)
    ans = ans + [mid] + ans[::-1]
    print(''.join(ans), file=sys.stderr)

File: server_data/contests/Contest3/replay/generator.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    srand(time(0));

    bool is_pal = rand() % 2;
    int n = rand() % 20 + 1;
    vector<int> cnts(26, 0);

    if (!is_pal) {
        for (int i = 0; i < n; ++i) {
            cnts[rand() % 26]++;
        }
    } else {
        for (int i = 0; i < n / 2; ++i) {
            cnts[rand() % 26] += 2;
        }
        if (n % 2 == 1) {
            cnts[rand() % 26]++;
        }
    }

    for (int i = 0; i < 26; ++i)
        cout << cnts[i] << ' ';
    cout << endl;

    string ans = "", mid = "";
    for (int i = 0; i < 26; ++i) {
        if (cnts[i] % 2 == 1) {
            if (!mid.empty()) {
                cerr << -1 << endl;
                return 0;
            }
            mid = string(1, 'a' + i);
        }
        ans += string(cnts[i] / 2, 'a' + i);
    }

    random_device rd;
    mt19937 g(rd());
    shuffle(ans.begin(), ans.end(), g);

    string final_str = ans + mid + string(ans.rbegin(), ans.rend());
    cerr << final_str << endl;

    return 0;
}
